src/main/java/HttpServer/ResponseWriter.java

```diff
@@ -24,37 +21,42 @@ public void write() throws IOException {
         System.out.println("\nWriting response:");
         System.out.println("--------------------------");
 
-        String statusLine = computeStatusLine(request);
-        writeLine(statusLine);
+        Response basicResponse = respondWithStatus(request);
 
-        String contentType = "Content-Type: text/html";
-        writeLine(contentType);
+        String statusLine = basicResponse.statusLine();
+        writeLine(statusLine);
 
-        String contentLength = "Content-Length: 0";
-        writeLine(contentLength);
+        if (basicResponse.containsKey("Content-Type")) {
+            String contentType = basicResponse.get("Content-Type");
+            writeLine("Content-Type: " + contentType);
+        }
 
-        System.out.println("<CRLF>");
-        writing.writeLine("");
+        if (basicResponse.containsKey("Body")) {
```

Comment:
I would extract a function responsible for writing the body.

Reply:
:white_check_mark: